FreeSurfer/LongitudinalFreeSurferPipeline.sh

```diff
@@ -0,0 +1,418 @@
+  else
+    log_Err_Abort "Cannot tell which version of FreeSurfer you are using."
+  fi
+
+  # strip out extraneous stuff from FreeSurfer version string
+  freesurfer_version_string_array=(${freesurfer_version_string//-/ })
+  freesurfer_version=${freesurfer_version_string_array[5]}
+  freesurfer_version=${freesurfer_version#v} # strip leading "v"
+
+  log_Msg "INFO: Determined that FreeSurfer version is: ${freesurfer_version}"
+
+  # break FreeSurfer version into components
+  # primary, secondary, and tertiary
+  # version X.Y.Z ==> X primary, Y secondary, Z tertiary
+  freesurfer_version_array=(${freesurfer_version//./ })
+
+  freesurfer_primary_version="${freesurfer_version_array[0]}"
+  freesurfer_primary_version=${freesurfer_primary_version//[!0-9]/}
+
+  freesurfer_secondary_version="${freesurfer_version_array[1]}"
+  freesurfer_secondary_version=${freesurfer_secondary_version//[!0-9]/}
+
+  freesurfer_tertiary_version="${freesurfer_version_array[2]}"
+  freesurfer_tertiary_version=${freesurfer_tertiary_version//[!0-9]/}
+
+  if [[ $(( ${freesurfer_primary_version} )) -lt 6 ]]; then
+    # e.g. 4.y.z, 5.y.z
+    log_Err_Abort "FreeSurfer version 6.0.0 or greater is required. (Use FreeSurferPipeline-v5.3.0-HCP.sh if you want to continue using FreeSurfer 5.3)"
+  fi
+}
+
+# Configure the use of FreeSurfer v6 custom tools
+configure_custom_tools
+
+# Show tool versions
+show_tool_versions
+
+# Validate version of FreeSurfer in use
+validate_freesurfer_version
+
+# ----------------------------------------------------------------------
+log_Msg "Starting main functionality"
+# ----------------------------------------------------------------------
+
+
+# ----------------------------------------------------------------------
+# Log values retrieved from positional parameters
+# ----------------------------------------------------------------------
+log_Msg "StudyFolder: ${StudyFolder}"
+log_Msg "SubjectID: ${SubjectID}"
+log_Msg "Sessions: ${Sessions}"
+log_Msg "TemplateID: ${TemplateID}"
+log_Msg "recon_all_seed: ${recon_all_seed}"
+log_Msg "extra_reconall_args_base: ${extra_reconall_args_base[*]+"${extra_reconall_args_base[*]}"}"
+log_Msg "extra_reconall_args_long: ${extra_reconall_args_long[*]+"${extra_reconall_args_long[*]}"}"
+
+# ----------------------------------------------------------------------
+log_Msg "Preparing the folder structure"
+# ----------------------------------------------------------------------
+Sessions=`echo ${Sessions} | sed 's/@/ /g'`
+log_Msg "After delimiter substitution, Sessions: ${Sessions}"
+LongDIR="${StudyFolder}/${SubjectID}.long.${TemplateID}/T1w"
+mkdir -p "${LongDIR}"
+for Session in ${Sessions} ; do
+  Source="${StudyFolder}/${Session}/T1w/${Session}"
+  Target="${LongDIR}/${Session}"
+  log_Msg "Creating a link: ${Source} => ${Target}"
+  ln -sf ${Source} ${Target}
+done
+
+
+if (( generate_timepoints_only != 1 )); then 
```

Comment:
@coalsont : Do we have a preference for 1/0 vs. TRUE/FALSE?

Reply:
My preference is to use `opts_StringToBool` for the user input handling, which handles both of those and yes/no, and returns 0/1.  Then I use `if ((generate_timepoints))` or `if ((! generate_timepoints))`, without explicit constants.

https://github.com/Washington-University/HCPpipelines/blob/9cfa37a378dff26fd5aa4fd2aad24d0136830f0f/global/scripts/RSNregression.sh#L72
https://github.com/Washington-University/HCPpipelines/blob/9cfa37a378dff26fd5aa4fd2aad24d0136830f0f/global/scripts/RSNregression.sh#L211

This does exactly what it looks like, and 1 (technically, any nonzero value, including negatives) means true, as expected.

Reply:
However, note that `(())` is a math expression context, and doesn't accept string values (which is why you don't need `$`), so something like `if ((x == TRUE))` isn't going to do what you expect (interpret TRUE as a variable name, probably undefined, but without `set -u`, undefined is equivalent to empty string, which math context reinterprets as 0, for...reasons).